code/modules/research/designs/weapon_designs.dm

```diff
@@ -23,13 +23,14 @@
 	category = list("Firing Pins")
 
 /datum/design/stunrevolver
-	name = "Stun Revolver"
+	name = "Shock Revolver"
 	desc = "A high-tech revolver that fires internal, reusable taser cartridges in a revolving cylinder. The cartridges can be recharged using conventional rechargers."
 	id = "stunrevolver"
-	req_tech = list("combat" = 3, "materials" = 3, "powerstorage" = 2)
+	req_tech = list("combat" = 3, "materials" = 3, "powerstorage" = 5)
 	build_type = PROTOLATHE
-	materials = list(MAT_METAL = 4000, MAT_GLASS = 1000)
-	build_path = /obj/item/weapon/gun/energy/stunrevolver
+	materials = list(MAT_METAL = 10000, MAT_GLASS = 10000, MAT_SILVER = 10000)
```

Comment:
Unless I am mistaken horribly, thats 5 silver sheets, which is ~2 ore veins of silver. Silver is incredibly common.

The gun is an AoE stun, I honestly might have priced it a bit cheap. I think the pins will be the big limit here rather than the resources.